x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/data_stream/131_sort_segments_migrate_to_data_stream.yml

```diff
@@ -30,14 +30,14 @@
   - do:
       index:
         index: my_ds
-        body: { "foo": "bar1", "@timestamp": "2021-08-01" }
+        body: { "foo": "bar1", "@timestamp": "2021-08-21" }
```

Comment:
The test failure appears to be dependent on the locale (en-MT) and possibly the JDK version in the reproduction line. Changing the sample dates to ones in which the month and day cannot be validly swapped (i.e., is `2021-08-01` interpreted as `Jan 08 21` or `Aug 01 21`) eliminates the test failure, but I'm not sure if that's the correct fix or just masking the problem.